processor/src/utils.rs

```diff
@@ -6,6 +6,100 @@ pub use vm_core::utils::*;
 
 use super::Felt;
 
+// FALCON SIGNATURE
+// ================================================================================================
+
+/// Event ID for pushing a Falcon signature to the advice stack.
+/// This event is used for testing purposes only.
+#[cfg(any(test, feature = "testing"))]
+pub const EVENT_FALCON_SIG_TO_STACK: u32 = 3419226139;
+
+/// Gets as input a vector containing a secret key, and a word representing a message and outputs a
+/// vector of values to be pushed onto the advice stack.
+///
+/// The values are the ones required for a Falcon signature verification inside the VM and they are:
+///
+/// 1. The challenge point, a tuple of elements representing an element in the quadratic extension
+///    field, at which we evaluate the polynomials in the subsequent three points to check the
+///    product relationship.
+/// 2. The expanded public key represented as the coefficients of a polynomial of degree < 512.
+/// 3. The signature represented as the coefficients of a polynomial of degree < 512.
+/// 4. The product of the above two polynomials in the ring of polynomials with coefficients in the
+///    Miden field.
+/// 5. The nonce represented as 8 field elements.
+///
+/// # Return
+///
+/// Returns a vector of field elements to be pushed onto the advice stack, or `None` if the secret
+/// key is malformed due to either incorrect length or failed decoding.
+#[cfg(feature = "std")]
+pub fn falcon_sign(sk: &[Felt], msg: vm_core::Word) -> Option<Vec<Felt>> {
```

Comment:
I chose to export this here still to facilitate the implementation of event handlers e.g. by `miden-base` or other libraries.

Reply:
We can't do that unfortunately because this function is needed by `push_falcon_signature()`, which is called by `DefaultHost::on_event`.

I think this is an artifact of `DefaultHost` being the `Host` used in tests, but also exported in a non-test context in `miden-processor`. For example, we would then be able to move this to `miden-stdlib` if `DefaultHost` was defined in `test-utils` instead.

So either
1. we leave as is
2. we move `DefaultHost` to `test-utils`, in which case there is no default implementation of `Host` exported anymore
3. We change `Test` to use some new `TestHost` defined in `test-utils`, which would allow us to move this to `miden-stdlib`

I think (3) is preferable, but would be better in another PR in case changing to a new `TestHost` ends up being a big-ish diff.

Reply:
Agreed that option 3 is the way to go. Could you take a look at how much work adding a `TestHost` could be? Maybe it could be a pretty simple wrapper around the `DefaultHost`? If it's not too much work, doing it this PR (or a PR stacked on top of this) would be better as moving this function to `miden-stdlib` later will be a breaking change.

Reply:
Good call, ended up being a small change. Created `TestHost` which wraps `DefaultHost`, cleaned up `DefaultHost::on_event()`, and moved `falcon_sign` and the event constant to `miden-stdlib`